src/lightning/pytorch/loops/epoch/training_epoch_loop.py

```diff
@@ -246,7 +245,7 @@ def on_advance_end(self) -> None:
         should_check_val = self._should_check_val_fx()
         if should_check_val:
             self.trainer.validating = True
-            self._run_validation()
+            self.val_loop.run()
```

Comment:
does this now handle validation loader reloading?

Reply:
Yes. The first thing `loop.run` does is `setup_data()`